karabo_data/geometry2.py

```diff
@@ -58,16 +63,18 @@ def snap(self):
         fs_vec = np.around(self.fs_vec[:2]).astype(np.int32)
         assert {tuple(np.abs(ss_vec)), tuple(np.abs(fs_vec))} == {(0, 1), (1, 0)}
         # Convert xy coordinates to yx indexes
-        return GridGeometryFragment(corner_pos[::-1], ss_vec[::-1], fs_vec[::-1])
+        return GridGeometryFragment(corner_pos[::-1], ss_vec[::-1], fs_vec[::-1],
+                                    self.ss_pixels, self.fs_pixels)
 
-class GridGeometryFragment:
-    ss_pixels = 64
-    fs_pixels = 128
 
+class GridGeometryFragment:
     # These coordinates are all (y, x), suitable for indexing a numpy array.
-    def __init__(self, corner_pos, ss_vec, fs_vec):
+    def __init__(self, corner_pos, ss_vec, fs_vec, ss_pixels, fs_pixels):
         self.ss_vec = ss_vec
         self.fs_vec = fs_vec
+        self.ss_pixels = ss_pixels
+        self.fs_pixels = fs_pixels
+
         if fs_vec[0] == 0:
```

Comment:
Isn't fs_vec[0] is 0 always ?

`assert {tuple(np.abs(ss_vec)), tuple(np.abs(fs_vec))} == {(0, 1), (1, 0)}`

`return GridGeometryFragment(corner_pos[::-1], ss_vec[::-1], fs_vec[::-1], self.ss_pixels, self.fs_pixels)`

Reply:
I'll add a comment to make that clearer - those are sets, so it's checking that between `fs_vec` and `ss_vec` there's one `(0, 1)` and one `(1, 0)`, but it doesn't assert which is which.

At present, LPD always has `fs_vec` in the x dimension, so it doesn't need transposing, and AGIPD always has it in the y dimension, so it does.

Reply:
(But the latter could change if AGIPD was assembled in the 'windmill' layout which I've heard is a possibility)